### PR TITLE
Component-Library storybook introduction fix

### LINKED ISSUE #19329: Fix component-library storybook introduction MDX render
### Description
The introduction to the component-library MDX document isn't rendering correctly.

Screenshot 2023-05-29 at 10 32 23 AM

### Technical Description
- Fix MDX document rendering `ui/components/component-library/COMPONENT-LIBARY.stories.mdx`

### Acceptance Criteria 
- Introduction document renders correctly

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- This PR closes #19329 
The Component-library introduction story is fixed and it maintains consistency with other introduction stories.

### Before

image

### After

image

## Manual Testing Steps
Pull this branch
run `yarn storybook`
check `Introduction` in `ComponentLibrary`

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by dhruvv173** [comment]
hey @georgewrmarshall, could you please review this PR?
Thanks!

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @dhruvv173, this looks great. Can you please update the name `COMPONENT-LIBRARY.stories.mdx` to `README.stories.mdx` we still want a readme and I'm hoping it will still show up on GitHub ok so there is still readable documentation about the component-library without having to run storybook

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, this looks great. Can you please update the name `COMPONENT-LIBRARY.stories.mdx` to `README.stories.mdx` we still want a readme and I'm hoping it will still show up on GitHub ok so there is still readable documentation about the component-library without having to run storybook

I renamed the file but it does not show up on GitHub as a `README`, should I make a `README.md` and paste the content there aswell?

**Comment by dhruvv173** [comment]
hey @georgewrmarshall, I did some research and found the fix to this. Now, it can be read on GitHub with `README.md` and on Storybook with `COMPONENT-LIBRARY.stories.mdx`. Attaching screenshots below from latest commit for reference.
I renamed the stories file as well, could you please check this once?

image
image

File: ui/components/component-library/README.md [hunk 1]
```diff
@@ -36,4 +36,4 @@ We are currently in the process of migrating all component-library components to
 
 If internal folks have any questions please reach out the design system team via the internal slack channel [#metamask-design-system](https://consensys.slack.com/archives/C0354T27M5M) 💁
 
-[DS components figma file](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/DS-Components?node-id=16%3A6) (internal)
+[DS components figma file](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/DS-Components?node-id=16%3A6) (internal)
```
Comment (georgewrmarshall) [inline ui/components/component-library/README.md]: Oh we made our Figma UI Kit publicly viewable so we can remove "(internal)"
```suggestion
[DS components figma file](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/DS-Components?node-id=16%3A6)
```

**Review by georgewrmarshall — DISMISSED** [review]
Hey @dhruvv173, checked out the branch looks good made a couple small updated wile I was there. Thanks for your contribution!

**Review by georgewrmarshall — APPROVED** [review]
Ah missed the rename of the file. Nice work

**Review by garrettbear — APPROVED** [review]

**Pull request merged**